clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -1640,8 +1640,18 @@ void CodeGenFunction::GenerateCode(GlobalDecl GD, llvm::Function *Fn,
            !getLangOpts().CUDAIsDevice &&
            FD->hasAttr<CUDAGlobalAttr>())
     CGM.getCUDARuntime().emitDeviceStub(*this, Args);
-  else if (isa<CXXMethodDecl>(FD) &&
-           cast<CXXMethodDecl>(FD)->isLambdaStaticInvoker()) {
+  else if (getLangOpts().CUDA && !getLangOpts().CUDAIsDevice &&
+           getLangOpts().SYCLIsHost && !FD->hasAttr<CUDAHostAttr>() &&
+           FD->hasAttr<CUDADeviceAttr>()) {
+    // Generate a dummy __host__ function for compiling CUDA sources in SYCL.
+    Fn->setLinkage(llvm::Function::WeakODRLinkage);
+    if (FD->getReturnType()->isVoidType())
+      Builder.CreateRetVoid();
+    else
+      Builder.CreateRet(llvm::UndefValue::get(Fn->getReturnType()));
+    return;
+  } else if (isa<CXXMethodDecl>(FD) &&
```

Comment:
```suggestion
    return;
  }
  if (isa<CXXMethodDecl>(FD) &&
```
https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return